File: Java/6.38.java
/*
A contact list is a place where you can store a specific contact with other associated information such as a phone number, email address, birthday, etc. Write a program that first takes as input an integer N that represents the number of word pairs in the list to follow. Word pairs consist of a name and a phone number (both strings). That list is followed by a name, and your program should output the phone number associated with that name. Assume that the list will always contain less than 20 word pairs.

Ex: If the input is:

3 Joe 123-5432 Linda 983-4123 Frank 867-5309 
Frank
the output is:

867-5309
Your program must define and call the following method. The return value of getPhoneNumber() is the phone number associated with the specific contact name.
*/

import java.util.Scanner; 

public class LabProgram {
   
   /* Define your method here */
   public static String getPhoneNumber(String[] nameVec, String[] phoneNumberVec, String contactName, int arraySize) {
      String phoneNumber = "";
      for (int i = 0; i < arraySize; i++) {
         if (contactName.equals(nameVec[i])) {
            phoneNumber = phoneNumberVec[i];
         }
      }
      return phoneNumber;
   }
   
   public static void main(String[] args) {
      /* Type your code here. */
      Scanner scan = new Scanner(System.in);
      String[] inputNames = new String[20];
      String[] inputNumbers = new String [20];
      int totalContacts;
      String searchName, numberResult;
      
      totalContacts = scan.nextInt();
      
      for (int i = 0; i < totalContacts; i++) {
         inputNames[i] = scan.next();
         inputNumbers[i] = scan.next();
      }
      
      searchName = scan.next();
      
      numberResult = getPhoneNumber(inputNames, inputNumbers, searchName, totalContacts);
      
      System.out.printf("%s\n", numberResult);
      
   }
}

File: CPP/6.38.cpp
/*
Write a program that reads a list of words. Then, the program outputs those words and their frequencies. 
The input begins with an integer indicating the number of words that follow.

Ex: If the input is:

5 hey hi Mark hi mark
the output is:

hey 1
hi 2
Mark 1
hi 2
mark 1
*/

#include <iostream>
#include <vector> 
#include <string>
using namespace std;

/* Define your function here */
int GetFrequencyOfWord(vector<string> wordsList,string currWord) {
   int counter=0;
   for(int i=0;i<wordsList.size();++i) {
      if(wordsList[i]==currWord)
      ++counter;
   }
   return counter;
}

int main() {
   /* Type your code here */
   int n, i;
   vector<string> wordsList;
   string word;

   cin>>n;
   for(i=0;i<n;++i) {
      cin >> word;
      wordsList.push_back(word); //add word to wordslist
   }

   for(i=0;i<n;++i) {
      cout << wordsList[i] << " " << GetFrequencyOfWord(wordsList,wordsList[i]) << endl;
   }

   return 0;
}
